Example for X.509 direct grant flow authentication (#30203)
keycloak/keycloak: docs/documentation/server_admin/topics/authentication/x509.adoc

[source,bash,subs="attributes+"]
----
curl \
  -d "client_id=resource-owner" \
  -d "client_secret=40cc097b-2a57-4c17-b36a-8fdf3fc2d578" \
  -d "grant_type=password" \
  --cacert /tmp/truststore.pem \
  --cert /tmp/keystore.pem:kssecret \
  "https://localhost:8543/realms/test/protocol/openid-connect/token"
----

The file `/tmp/truststore.pem` points to the file with the truststore containing the certificate of the {project_name} server. The file `/tmp/keystore.pem` contains
the private key and certificates corresponding to the {project_name} user, which would be successfully authenticated by this request. It is dependent on the configuration of the authenticator on how
exactly is the content from the certificate mapped to the {project_name} user as described in <<_x509-config, the configuration section>>. The `kssecret` might be the password of this keystore file.

According to your environment, it might be needed to use more options to CURL commands like for instance:

* Option `--insecure` if you are using self-signed certificates

> * Option `--capath` for include the whole directory containing the certificate authority path

Reviewer: ```suggestion
* Option `--capath` to include the whole directory containing the certificate authority path
```

* Options `--cert-type` or `--key-type` in case you want to use different files than `PEM`

> Please consult the documentation of the `curl` tool for the details if needed. If you are using other tool than `curl`, you may need to
> consult the documentation of your tool, but the setup would be similar. There is a need to include keystore and truststore as well as client credentials in case you are using confidential

Reviewer: ```suggestion
Please consult the documentation of the `curl` tool for the details if needed. If you are using other tools than `curl`, 
consult the documentation of your tool. However, the setup would be similar.  A need exists to include keystore and truststore as well as client credentials in case you are using confidential
```
